junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java

```diff
@@ -59,6 +65,8 @@ class DefaultLauncher implements Launcher {
 			"PostDiscoveryFilter array must not contain null elements");
 		this.discoveryOrchestrator = new EngineDiscoveryOrchestrator(testEngines,
 			unmodifiableCollection(postDiscoveryFilters), listenerRegistry.launcherDiscoveryListeners);
+		this.launcherSession = LauncherFactory.openSession(config);
```

Comment:
We should not open a session here but pass in the `sessionStore` to the constructor.